test/karma/tail.js

```diff
@@ -0,0 +1,15 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+// Stop the default init functions running to avoid conflicts in tests
```

Comment:
The listeners are injected in the global scope as we can't call them direct (something's got to kick everything off). Therefore using dependency injection is difficult. I think this method is slightly better than stubbing addEventListener globally.